air/src/options.rs

```diff
@@ -218,6 +218,12 @@ impl Default for ExecutionOptions {
 }
 
 impl ExecutionOptions {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+
+    /// The maximum number of VM cycles a transaction is allowed to take.
+    const MAX_TX_EXECUTION_CYCLES: u32 = 1 << 30;
```

Comment:
I would remove `TX` from the name, and maybe also just call it `MAX_CYCLES`.

Also, I'd set it to $2^{29}$.

Lastly, we need to make sure that we validate this in the constructor and during deserialization (if any).